### ISSUE TITLE
ZobristKey is not equal to ZobristKey of FEN

### ISSUE THREAD

**Issue by CoolJWB** [SebLague/Chess-Challenge#227]
Maybe this is expected behavior but: it seems as if the ZobristKey of the current board position does not match the ZobristKey of a FEN board of that same position. It is not an issue of board.GetFenString(), I tried with an independent FEN aswell.

```c#
// How to replicate on version 1.12:

public Move Think(Board board, Timer timer)
{
  if(board.ZobristKey == Board.CreateBoardFromFEN(board.GetFenString()).ZobristKey)
  {
      Console.WriteLine("True");
  }
  else
  {
      Console.WriteLine("False");
  }

// ...
```

![image](https://github.com/SebLague/Chess-Challenge/assets/13751243/385ba592-dfed-43f5-8498-1ee4f93d8e62)

**Comment by mcthouacbb** [comment]
What does it print if you change the condition to "board.GetFenString() == Board.CreateBoardFromFEN(board.GetFenString()).GetFenString()"?

**Comment by CoolJWB** [comment]
True all the time, 

> What does it print if you change the condition to "board.GetFenString() == Board.CreateBoardFromFEN(board.GetFenString()).GetFenString()"?

True all the time.

**Comment by SebLague** [comment]
Fixed in v1.13 Thanks for letting me know, and sorry for the bug!

**Comment by CoolJWB** [comment]
> Fixed in v1.13 Thanks for letting me know, and sorry for the bug!

All good, and thanks for solving it :)
